torch/csrc/jit/tracer.cpp

```diff
@@ -89,4 +89,36 @@ void nontraceableBackwardSubgraph(const variable_list& inputs, const variable_li
   std::make_shared<autograd::Eval>()->replaceSubgraph(inputs, outputs);
 }
 
+Node* recordTraceWorker(std::string op, // TODO: make this a Symbol
+                        at::ArrayRef<Variable> inputs,
+                        at::ArrayRef<Variable> outputs) {
+  auto state = getTracingState(inputs);
+  auto& graph = state->graph;
+  // TODO: Technically, we could reduce the scope of the lock, but since we
+  // haven't actually specified what the locking contract is, be conservative.
+  auto state_lock = state->lock();
+
+  Node *n = graph->create(stringToSymbol(op));
+  for (Variable input : inputs) {
+    graph->addInput(getValueTrace(state, input));
+  }
+
+  // NB: Order matters. This must append after inputs but before outputs.
+  graph->appendNode(n);
+
+  int i = 0;
+  for (Variable output : outputs) {
+    Node* sel = graph->appendNode(graph->createSelect(n, i));
```

Comment:
IIRC we're going to get rid of ONNXification pass for these "native" nodes, so maybe let's only create selects for ops that actually have multiple outputs in here?

Reply:
I guess this would make the ONNX pass (for export) marginally more complicated, but we might end up doing this anyway when we start fixing the JIT for the new world order.

Reply:
Oh I wasn't thinking about the export pass, but I just realized that we'll be getting rid of the current "initial graph translation" pass. In this case, I think we should definitely now use Selects if node has only a single output. There will be no pass to fix this, and it's trivial to implement here

Reply:
OK

Reply:
OK, I tried doing this, and there's another pre-req, which is managing `Handle` outputs. I'd prefer not to touch this until I am getting the JIT working again, so can we delay this for later?

Reply:
Handles are only for non-traceable things, but all C++ functions should be traceable

Reply:
Yes. My point is that I have to write another version of the ONNX export codepath that handles not-multiple-outputs and I want to merge the current set of changes first (updated PR coming)

Reply:
Alright, let's just unbreak ONNX asap